### PR TITLE
Adding BigtableInstanceName.

### PR DESCRIPTION
**Pull request by sduskis**

### CODE REVIEW

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableInstanceName.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+ * This class encapsulates a Bigtable instance name.  An instance name is of the form
+ * projects/(projectId)/instances/(instanceId).  It also has convenience methods
+ * to create a tableName and a tableId.  TableName is (instanceName)/tables/(tableId).
+ */
+public class BigtableInstanceName implements Serializable {
+  private static final long serialVersionUID = 1L;
+
+  public static final String BIGTABLE_V1_INSTANCE_FMT = "projects/%s/instances/%s";
```
Comment (garye) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableInstanceName.java]: BIGTABLE_V1_INSTANCE_FMT -> BIGTABLE_V2_INSTANCE_FMT?

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableInstanceName.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  public static final String TABLE_SEPARATOR = "tables";
+
+  private final String InstanceName;
+
+  public BigtableInstanceName(String projectId, String instanceId) {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(projectId), "projectId must be supplied");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(instanceId), "instnaceId must be supplied");
```
Comment (garye) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableInstanceName.java]: instnaceId -> instanceId

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableInstanceName.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ * projects/(projectId)/instances/(instanceId).  It also has convenience methods
+ * to create a tableName and a tableId.  TableName is (instanceName)/tables/(tableId).
+ */
+public class BigtableInstanceName implements Serializable {
+  private static final long serialVersionUID = 1L;
+
+  public static final String BIGTABLE_V1_INSTANCE_FMT = "projects/%s/instances/%s";
+  public static final String TABLE_SEPARATOR = "tables";
+
+  private final String InstanceName;
```
Comment (garye) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableInstanceName.java]: InstanceName -> instanceName?

**Comment by sduskis** [comment]
I addressed the typos.  Sorry about them.  PTAL.

**Comment by garye** [comment]
LGTM

**Pull request merged**